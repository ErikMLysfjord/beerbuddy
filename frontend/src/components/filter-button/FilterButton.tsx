import { FilterFilled } from "@ant-design/icons";
import { Button, Modal } from "antd";
import styles from "./FilterButton.module.css";
import { useState } from "react";
import Filters from "../filters/Filters";
import useFetchMoreBeers from "../../utils/useFetchMoreBeers";

const FilterButton = () => {
  const [showFilter, setShowFilter] = useState(false);
  const { fetchMore } = useFetchMoreBeers();

  const handleClick = () => {
    setShowFilter(!showFilter);
  };

  return (
    <div className={styles.filterButtonWrapper}>
      <Button
        type="primary"
        icon={<FilterFilled />}
        onClick={handleClick}
      ></Button>
      <Modal
        title="Filters"
        open={showFilter}
        onCancel={handleClick}
        footer={null}
        width={1000}
      >
        <Filters fetchMore={fetchMore} />
      </Modal>
    </div>
  );
};
export default FilterButton;
