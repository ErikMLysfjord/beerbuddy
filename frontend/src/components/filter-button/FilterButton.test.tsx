import { describe, it, expect } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import FilterButton from "./FilterButton";
import { act } from "react-dom/test-utils";

// Fixes a warning which seems to be a bug.
// https://github.com/nickcolley/jest-axe/issues/147
const { getComputedStyle } = window;
window.getComputedStyle = (elt) => getComputedStyle(elt);

describe("FilterButton", () => {
  it("renders correctly", () => {
    const { container } = render(<FilterButton />);
    expect(container).toMatchSnapshot();
  });

  it("should open modal", () => {
    const { getAllByRole } = render(<FilterButton />);

    expect(screen.queryByText("Filters")).not.toBeInTheDocument();

    act(() => {
      getAllByRole("button")[0].click();
    });

    expect(screen.queryByText("Filters")).toBeInTheDocument();
  });

  it("should close modal", () => {
    render(<FilterButton />);

    expect(screen.queryByText("Filters")).not.toBeInTheDocument();

    fireEvent.click(screen.getByRole("button"));

    const dialog = screen.getByRole("dialog");

    /* Need to test the visibility of the wrapper of the modal,
    since that is the element gaining the display: none on closing. */
    const modalWrapper = dialog.parentElement;
    expect(modalWrapper).toBeVisible();

    const closeButton = screen.getByRole("button", { name: /close/i });

    fireEvent.click(closeButton);
    expect(modalWrapper).not.toBeVisible();
  });
});
