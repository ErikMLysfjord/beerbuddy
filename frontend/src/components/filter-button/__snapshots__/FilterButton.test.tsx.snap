// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterButton > renders correctly 1`] = `
<div>
  <div
    class="_filterButtonWrapper_1f5523"
  >
    <button
      class="ant-btn css-dev-only-do-not-override-pr0fja ant-btn-primary ant-btn-icon-only"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          aria-label="filter"
          class="anticon anticon-filter"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="filter"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
            />
          </svg>
        </span>
      </span>
    </button>
  </div>
</div>
`;
